games-action/armagetronad/files/armagetronad-0.2.8.3.3-gcc6.patch

```diff
@@ -0,0 +1,222 @@
+Bug: https://bugs.gentoo.org/show_bug.cgi?id=617768
+Commit: https://sourceforge.net/p/armagetronad/code/10845/
+
+--- a/src/engine/eNetGameObject.cpp
++++ b/src/engine/eNetGameObject.cpp
+@@ -85,7 +85,7 @@
+ eNetGameObject::eNetGameObject(eGrid *grid, const eCoord &pos,const eCoord &dir,
+                                ePlayerNetID* p,bool autodelete)
+         :eGameObject(grid, pos,dir,NULL,autodelete),
+-nNetObject(p->Owner()),player(p){
++         nNetObject(Owner(p)),player(p){
+     lastClientsideAction=0;
+     if (sn_GetNetState()!=nCLIENT)
+         RequestSync();
+@@ -205,7 +205,7 @@
+ 
+ void eNetGameObject::SetPlayer(ePlayerNetID* a_player)
+ {
+-    tASSERT( !a_player || Owner() == player->Owner() );
++    tASSERT( !a_player || Owner() == Owner(player) );
+     player  = a_player;
+     if ( laggometerSmooth == 0 && sn_GetNetState() != nCLIENT )
+         laggometerSmooth = laggometer = se_GetPing( player );
+--- a/src/engine/ePlayer.cpp
++++ b/src/engine/ePlayer.cpp
+@@ -56,6 +56,7 @@
+ #include "nConfig.h"
+ #include "nNetwork.h"
+ #include <time.h>
++#include <climits>
+ 
+ int se_lastSaidMaxEntries = 8;
+ 
+@@ -1689,9 +1690,9 @@
+         // foo (Red Team) --> Blue Team: some message here
+         eTeam *senderTeam = sender->CurrentTeam();
+         console << tColoredString::ColorString(1,1,.5) << " (";
+-        console << *senderTeam;
++        console << senderTeam;
+         console << tColoredString::ColorString(1,1,.5) << ") --> ";
+-        console << *team;
++        console << team;
+     }
+ 
+     console << tColoredString::ColorString(1,1,.5) << ": ";
+@@ -1923,7 +1924,7 @@
+         else {
+             eTeam *senderTeam = sender->CurrentTeam();
+             say << tColoredString::ColorString(1,1,.5) << " (";
+-            say << *team;
++            say << team;
+             say << tColoredString::ColorString(1,1,.5) << " ) --> ";
+             say << senderTeam;
+         }
+@@ -3941,7 +3942,7 @@
+     sg_ClampPingCharity( ::pingCharity );
+ }
+ 
+-static int IMPOSSIBLY_LOW_SCORE=(-1 << 31);
++static int IMPOSSIBLY_LOW_SCORE=INT_MIN;
```

Comment:
We'll keep it (this obviously isn't your fault, but this is not the C++-y way of doing limits...)